logstash-core/src/main/java/org/logstash/execution/AbstractPipelineExt.java

```diff
@@ -540,6 +576,23 @@ private LongCounter initOrGetCounterMetric(final ThreadContext context,
         return retrievedMetric.toJava(LongCounter.class);
     }
 
+    private Optional<NumberGauge> initOrGetNumberGaugeMetric(final ThreadContext context,
+                                                             final RubySymbol[] subPipelineNamespacePath,
+                                                             final RubySymbol metricName) {
+        final IRubyObject collector = this.metric.collector(context);
+        final IRubyObject fullNamespace = pipelineNamespacedPath(subPipelineNamespacePath);
+        final IRubyObject retrievedMetric = collector.callMethod(context, "get", new IRubyObject[]{fullNamespace, metricName, context.runtime.newSymbol("gauge")});
+
+        LazyDelegatingGauge delegatingGauge = retrievedMetric.toJava(LazyDelegatingGauge.class);
+        if (Objects.isNull(delegatingGauge.getType()) ||
+                MetricType.GAUGE_NUMBER.asString().equals(delegatingGauge.getType().asString()) == false) {
```

Comment:
No need to route through `MetricType#toString()` and `String#equals()` to get equality between instances of the `MetricType` enum.

Reply:
Yeap, if it is pure Java makes sense. I was thinking if there is Jruby object behind, this would be safer.